src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1406,11 +1407,15 @@ def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_
                     f'`.{fn}(ckpt_path="best")` is set but `ModelCheckpoint` is not configured to save the best model.'
                 )
             # load best weights
-            ckpt_path = self.checkpoint_callback.best_model_path
+            ckpt_path = (
+                self.checkpoint_callback.best_model_path
+                if hasattr(self.checkpoint_callback, "best_model_path")
+                else None
+            )
```

Comment:
```suggestion
            ckpt_path = getattr(self.checkpoint_callback, "best_model_path", None)
```

Reply:
Oh my, I feel dumb for using this antipattern. Thanks @awaelchli for pointing it out :)